code/iaas/allocator/src/main/java/io/cattle/platform/allocator/dao/impl/AllocatorDaoImpl.java

```diff
@@ -591,6 +596,35 @@ private void updateInstancePorts(List<Map<String, Object>> dataList) {
         return;
     }
     
+    @SuppressWarnings("unchecked")
+	private void updateInstancePortsForLoadBalancer(List<Map<String, Object>> dataList, List<Port> newPorts) {
```

Comment:
I'd change the name for this method. It should not be load balancer specific. Why don't we just put all the logic to updateInstanceForNewPorts